### PR TITLE
Update SG notebook

### PR DESCRIPTION
**Pull request by naimnv**
Update SG notebook to handle a handle a couple of cases where the current notebook would fail for unweighted graph and  if the graph doesn't  have node id 0.

**Changes for SG notebook:**
- For graphs without weights, run bfs instead of sssp.
- For the input graph doesn't have node with id 0, peek an existing node id as seed for BFS and SSP
- simplify bookkeeping for run statistics

**Changes for MG notebook:**
-  Call SSSP or BFS based on weighted attribute

### CODE REVIEW

**Review by BradReesWork — APPROVED** [review]

File: notebooks/cugraph_benchmarks/synth_release.ipynb [hunk 1]
```diff
@@ -40,7 +40,8 @@
     "| Author        |    Date    |  Update             | cuGraph Version |  Test Hardware         |\n",
     "| --------------|------------|---------------------|-----------------|------------------------|\n",
     "| Don Acosta    | 1/12/2023  | Created             | 23.02 nightly   | RTX A6000, CUDA 11.7   |\n",
-    "| Brad Rees     | 1/27/2023  | Modified            | 23.02 nightly   | RTX A6000, CUDA 11.7   |\n"
+    "| Brad Rees     | 1/27/2023  | Modified            | 23.02 nightly   | RTX A6000, CUDA 11.7   |\n",
+    "| Naim, Md      | 2/12/2024  | Modified            | 24.04 nightly   | RTX A6000, CUDA 12.0   |\n"
```
Comment (acostadon) [inline notebooks/cugraph_benchmarks/synth_release.ipynb:44]: This should say 2xA6000 if that is what you used to test multi-GPU.

**Review by acostadon — CHANGES_REQUESTED** [review]
Changes look great
Did you test this on multi-GPU, if so please note what GPU's were used in the header.

**Review by acostadon — APPROVED** [review]
Looks great

**Comment by naimnv** [comment]
> Changes look great Did you test this on multi-GPU, if so please note what GPU's were used in the header.

For `synth_release_single_node_multi_gpu.ipynb`  the hardware is now `2xRTX A600`
For synth_release_single_gpu.ipynb the hardware is  `RTX A600`

**Pull request merged**